data_model/src/account.rs

```diff
@@ -207,8 +196,7 @@ impl NewAccount {
     Debug,
     Display,
     Clone,
-    PartialEq,
-    Eq,
+    OrdEqHash,
```

Comment:
That's can be dangerous, we should think twice here. Account is used in a lot of quires, instrucitons and collections. I'm not sure if we want `AccountA` be equal to `AccountB` if they have the same id. This is not how it was before.

Reply:
The same thing for other structs where previously `Eq` was using all fields and now only uses `id`. I don't think that's a good idea. If we want check for *id* equality we can just `account_a.id() == account_b.id()`.

Reply:
Considering that account id contains domain id, why is this not a good idea?

Reply:
I don't understand your example... I just mean that we should probably use all fields to check accounts for equality, not just id

Reply:
I just wanted to know the reason why we should require structural vs nominal equality for structs that are uniquely identifiable, i.e. that implement `Identifiable`

Reply:
Shouldn't the id always be unique and thus sufficient? I see that the only case where id fields can be same, but some other fields different is when two accounts come from different domains -- but then they probably shouldn't be compared like that in the first place. So comparison of all fields would only give us a way to catch illegal duplicates, no?

Reply:
Yes, I'm thinking about something like illegal duplicates. I just had a bad experience with PermissionToken when it had a bug in its comparison traits and now I'm afraid of the same thing here.
But ok, maybe it's not that dangerous. If all our tests will pass then I have no claims

Reply:
@Arjentix I get where you're coming from, though `PermissionToken` was never `Identifiable`. If a type is `Identifiable`, which means it's `Id` is unique to the blockchain, then there should be no reason why not to use this optimization.

The only moment that I can think of when you would produce a duplicate is for `Registrable`s (should they be `Identifiable` at all?), when trying to register a new `Identifiable` like `Domain` with the same `Id`. But we reject these transactions immediately